### PR TITLE
Configure smart_proxy_ansible plugin

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: manifests/plugin/ansible.pp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# = Ansible proxy plugin
+#
+# This class installs Ansible support for Foreman proxy
+#
+# === Parameters:
+#
+# $ansible_dir:: Ansible directory to search for available roles
+#
+# $working_dir:: A directory where the playbooks will be generated.
+#                A tmp directory will be created when left blank
```
Comment (stbenjam) [inline manifests/plugin/ansible.pp]: I think both of these could go to `Advanced` as well.

File: manifests/plugin/ansible.pp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# $enabled::     Enables/disables the plugin
+#                type:boolean
+#
+# $listen_on::   Proxy feature listens on https, http, or both
+#
+class foreman_proxy::plugin::ansible (
+  $enabled     = $::foreman_proxy::plugin::ansible::params::enabled,
+  $listen_on   = $::foreman_proxy::plugin::ansible::params::listen_on,
+  $ansible_dir = $::foreman_proxy::plugin::ansible::params::ansible_dir,
+  $working_dir = $::foreman_proxy::plugin::ansible::params::working_dir,
```
Comment (ekohl) [inline manifests/plugin/ansible.pp]: Validate this behind an if statement?
```puppet
if $working_dir {
  validate_absolute_path($working_dir)
}
```

**Comment by iNecas** [comment]
Good calls, addressed. Thanks @ekohl and @stbenjam for review

**Comment by ekohl** [comment]
Thanks!

**Comment by ehelms** [comment]
@iNecas Is this going in 1.13.1 since the plugin is available in the plugin repositories?

**Comment by stbenjam** [comment]
It needs to get cherry-picked into the 4.0-stable branch and a new release made, not sure how the release process works for the Foreman puppet modules, @ekohl or @domcleal would maybe know, but smart_proxy_omaha is in the same situation AFAIK.

**Comment by iNecas** [comment]
I've filed http://projects.theforeman.org/issues/16791 as a request for cherry-picking the changes into 1.13.1 http://projects.theforeman.org/issues/16791, similar for infoblox http://projects.theforeman.org/issues/16790

**Pull request closed without merging**